tests/test_merge_dynamic_streams_block.py

```diff
@@ -0,0 +1,163 @@
+from collections import defaultdict
+from time import sleep
+from unittest.mock import MagicMock
+
+from nio.block.terminals import DEFAULT_TERMINAL
+from nio.signal.base import Signal
+from nio.testing.block_test_case import NIOBlockTestCase
+
+from ..merge_dynamic_streams_block import MergeDynamicStreams
+
+
+class TestMergeDynamicStreams(NIOBlockTestCase):
+
+    def process_test_signals(self, blk):
+        blk.process_signals([Signal({"A": "a", "stream": 1})])
+        sleep(0.2)
```

Comment:
I'm going to leave the sleeps because they make sense for this block, but I did manage to get what you are saying working

```
 with patch('{}.Job'.format(MergeDynamicStreams().__module__)) as job:
     blk.process_signals([Signal({"A": "a"})], input_id='input_1')
     if job.called:
         job.call_args[0][0](*job.call_args[0][-2:])
```